docs/operations/versioning.md

```diff
@@ -0,0 +1,305 @@
+Below is an example of a config file enabling `Version Management` for Kubernetes as well as Machine Image versions. 
+For each the configured expiration duration is 4 months. 
+This means that versions that are being deprecated by the `Version Management` will expire at the date of deprecation + 4 months.
+
+``` yaml
+apiVersion: controllermanager.config.gardener.cloud/v1alpha1
+kind: ControllerManagerConfiguration
+controllers:
+  cloudProfile:
+    concurrentSyncs: 5
+    kubernetesVersionManagement:
+      expirationDuration: 2880h # defaults to 120 days / 4 months
+    machineImageVersionManagement:
+      expirationDuration: 2880h # defaults to 120 days / 4 months
+```
+
+
+## Version Maintenance
+
+`Maintained` versions are commonly being referred to versions that have active development support for security updates and bug fixes. How many versions are being maintained is up to the individual project / product.
+For instance, Kubernetes releases **minor** versions roughly every three months and usually maintains **three minor** versions (the actual and the last two). 
+This means Kubernetes clusters having a version that is not `maintained` any more, run into the risk of having severe bugs or known security problems. 
+Considering Kubernetes minor versions: 1.18 (being the most recent), 1.17, 1.16, 1.15 and 1.14. Then versions 1.18, 1.17, 1.16 are `maintained` and versions 1.15 and 1.14 are `unmaintained`
+
+The operator of Gardener can optionally define `maintained` minor versions through the configuration file of the Gardener Controller Manager.
+This can be done by specifying the number of maintained minor versions starting from the most recent. Given the Kubernetes project example above, the amount of maintained minor versions is `3`.
+The configuration is located in the Gardener Controller Manager config at: `controllers.cloudProfile.kubernetesVersionManagement.versionMaintenance` and `controllers.cloudProfile.machineImageVersionManagement.versionMaintenance`.
+
+
+`Version Maintenance` helps with the handling of `maintained` and `unmaintained` versions, specifically:
+ - Enforces one `supported` version per `maintained` **minor version** (versus only one `supported` version without `Version Maintenance`)
+   - be aware that if there is no supported version for a maintained minor version, it will pick the **latest** version independent of the classification and sets it to `supported`.
+ - Deprecates all versions with an `unmaintained` minor (no matter what classification they have).
+ - Sets the correct deprecation date for `unmaintained` versions (`controllers.cloudProfile.kubernetesVersionManagement.expirationDuration`).
+ - Does not set an expiration date on the latest version with an `unmaintained` version. Otherwise, an undesired and automated forceful minor Kubernetes version upgrade is possible. This is potentially harmful for the deployed workload.  
```

Comment:
```suggestion
 - Does not set an expiration date on the latest version with an `unmaintained` version. Otherwise, an undesired and automated forceful minor Kubernetes version upgrade is possible. This is potentially harmful for the deployed workload. If a Gardener operator wants to phase out a minor version he/she has to set the expiration date manually.
```